randovania/games/cave_story/layout/cs_cosmetic_patches.py

```diff
@@ -0,0 +1,207 @@
+import dataclasses
+import logging
+from random import Random
+from enum import Enum, auto, unique
+from pathlib import Path
+from typing import Optional
+from randovania.bitpacking.bitpacking import BitPackDataclass, BitPackEnum
+from randovania.bitpacking.json_dataclass import JsonDataclass
+
+from randovania.games.game import RandovaniaGame
+from randovania.layout.base.cosmetic_patches import BaseCosmeticPatches
+
+@unique
+class MyChar(BitPackEnum, Enum):
+    QUOTE="Quote"
+    CURLY="Curly"
+    SUE="Sue"
+    TOROKO="Toroko"
+    KING="King"
+    CHACO="Chaco"
+    KANPACHI="Kanpachi"
+    MISERY="Misery"
+    FROG="Frog"
+    RANDOM="Random"
+    CUSTOM="Custom"
+
+    def mychar_bmp(self, rng: Random) -> Optional[Path]:
+        if self == MyChar.CUSTOM:
+            return None
+        if self == MyChar.RANDOM:
+            options = list(MyChar)
+            options.remove(MyChar.RANDOM)
+            options.remove(MyChar.CUSTOM)
+            return rng.choice(options).mychar_bmp(rng)
+        return str(RandovaniaGame.CAVE_STORY.data_path.joinpath(f"assets/mychar/{self.value}.bmp"))
+    
+    @property
+    def ui_icon(self) -> Path:
+        return RandovaniaGame.CAVE_STORY.data_path.joinpath(f"assets/icon/{self.value}.png")
+    
+    @property
+    def description(self) -> Optional[str]:
+        if self == MyChar.RANDOM:
+            return "Select a random MyChar for each seed."
+        if self == MyChar.CUSTOM:
+            return "Provide your own MyChar! Place it in your data folder."
+    
+    def next_mychar(self, reverse: bool) -> "MyChar":
+        upcoming = list(MyChar)
+        if reverse:
+            upcoming.reverse()
+        upcoming = upcoming[upcoming.index(self)+1:]
+        return next((mychar for mychar in upcoming), MyChar.CUSTOM if reverse else MyChar.QUOTE)
```

Comment:
Not really sure what this code is doing.

Could it just be `upcoming[(upcoming.index(self) + offset) % len(upcoming)]`, with offset being 1 or -1 if reverse? Reminder that `my_list[-1]` is the last element.

Reply:
:+1: